lib/src/common/disposable.dart

```diff
@@ -26,20 +26,41 @@ abstract class _Disposable {
   Future<Null> dispose();
 }
 
-class InternalDisposable implements _Disposable {
+class InternalDisposable implements _Disposable, ManagedDisposer {
```

Comment:
Are you saying rather than export an interface and implementing here, just create the implementation and export it?